pkg/server/devfile-handler.go

```diff
@@ -204,41 +212,32 @@ func getService(devfileObj parser.DevfileObj, filterOptions common.DevfileOption
 		return corev1.Service{}, err
 	}
 
+	portNumber, err := strconv.Atoi(imagePort)
+	if err != nil {
+		return corev1.Service{}, err
+	}
+
+	svcPort := corev1.ServicePort{
+		Name:       fmt.Sprintf("http-%v", imagePort),
+		Port:       int32(portNumber),
+		TargetPort: intstr.FromString(imagePort),
+	}
+	service.Spec.Ports = append(service.Spec.Ports, svcPort)
+
 	return *service, nil
 }
 
-func getRoutes(data devfileForm, containerComponents []devfilev1.Component) routev1.Route {
-
-	var routes []routev1.Route
-
-	for _, comp := range containerComponents {
-		for _, endpoint := range comp.Container.Endpoints {
-			if endpoint.Exposure == devfilev1.NoneEndpointExposure || endpoint.Exposure == devfilev1.InternalEndpointExposure {
-				continue
-			}
-			secure := false
-			if endpoint.Secure || endpoint.Protocol == "https" || endpoint.Protocol == "wss" {
-				secure = true
-			}
-			path := "/"
-			if endpoint.Path != "" {
-				path = endpoint.Path
-			}
-
-			routeParams := generator.RouteParams{
-				TypeMeta: generator.GetTypeMeta("Route", "route.openshift.io/v1"),
-				RouteSpecParams: generator.RouteSpecParams{
-					ServiceName: data.Name,
-					PortNumber:  intstr.FromInt(endpoint.TargetPort),
-					Path:        path,
-					Secure:      secure,
-				},
-			}
-
-			route := generator.GetRoute(routeParams)
-			routes = append(routes, *route)
-		}
-	}
-
-	return routes[0]
+func getRouteForDockerImage(data devfileForm, imagePort string) routev1.Route {
+
+	routeParams := generator.RouteParams{
+		TypeMeta: generator.GetTypeMeta("Route", "route.openshift.io/v1"),
+		RouteSpecParams: generator.RouteSpecParams{
+			ServiceName: data.Name,
+			PortNumber:  intstr.FromString(imagePort),
+			Path:        "/",
```

Comment:
I was thinking the same thing, if ODC is going to support 1+ images or 1+ routes. but that needs more discussion. 
since the PR is only for switching to use 2.2.0 devfile samples from registry, I think the current change is fine. 
we can always submit new PRs once we decided to support more in ODC.

Reply:
In addition, we should an issue to encourage odc to switch to consume the deploy commands and Kubernetes components. if that's the case, the routes and path will no longer be a problem.